### PR TITLE
[gronkh] Improve extractor

### LINKED ISSUE #3300: Gronkh.tv Unsupported URL
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.03.08.1**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Germany

### Description

Hi,

i have tested the site https://gronkh.tv/, the site was added with that pr https://github.com/yt-dlp/yt-dlp/pull/1299. But yt-dlp says "Unsupported URL". The Site is only working when i am using an specfic video url but not the full site.

### Verbose log

```shell
root@yt-dlp:/config# yt-dlp -vU https://gronkh.tv
[debug] Command-line config: ['-vU', 'https://gronkh.tv']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.03.08.2 [e491d06d3]
[debug] Python version 3.9.7 (CPython 64bit) - Linux-5.13.19-5-pve-x86_64-with
[debug] exe versions: ffmpeg 5.0 (setts), ffprobe 5.0
[debug] Optional libraries: sqlite
[debug] Proxy map: {}
Latest version: 2022.03.08.1, Current version: 2022.03.08.2
yt-dlp is up to date (2022.03.08.2)
[debug] [generic] Extracting URL: https://gronkh.tv
[generic] gronkh: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] gronkh: Downloading webpage
[generic] gronkh: Extracting information
[debug] Looking for video embeds
ERROR: Unsupported URL: https://gronkh.tv
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 1389, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 1459, in __extract_info
    ie_result = ie.extract(url)
  File "/usr/lib/python3.9/site-packages/yt_dlp/extractor/common.py", line 617, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3.9/site-packages/yt_dlp/extractor/generic.py", line 3998, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://gronkh.tv
```

### PR DESCRIPTION
**Pull request by hatienl0i261299**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Close #3300

### CODE REVIEW

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/gronkh.py [hunk 1]
```diff
@@ -41,3 +47,67 @@ def _real_extract(self, url):
...
+        'playlist_count': 16,
+    }]
+
+        def _entries(self):
+            for type_ in ('recent', 'views'):
+                info = self._download_json(
+                    f'https://api.gronkh.tv/v1/video/discovery/{type_}', 'feed', note=f'Downloading {type_} API JSON')
+                for item in traverse_obj(info, ('discovery', ...)) or []:
+                    yield self.url_result(
+                            f'https://gronkh.tv/watch/stream/{item.get("episode")}', GronkhIE, item.get('title'))
```
Comment (pukkandan) [inline yt_dlp/extractor/gronkh.py:81]:
```suggestion
    def _entries(self):
        for type_ in ('recent', 'views'):
            info = self._download_json(
                f'https://api.gronkh.tv/v1/video/discovery/{type_}', 'feed', note=f'Downloading {type_} API JSON')
            for item in traverse_obj(info, ('discovery', ...)) or []:
                yield self.url_result(
                        f'https://gronkh.tv/watch/stream/{item.get("episode")}', GronkhIE, item.get('title'))
```

File: yt_dlp/extractor/gronkh.py [hunk 2]
```diff
@@ -44,3 +50,73 @@ def _real_extract(self, url):
...
+    _VALID_URL = r'https?://(?:www\.)?gronkh\.tv/?(feed/?)?$'
+    IE_NAME = 'gronkh:feed'
+
+    _TESTS = [{
+        'url': 'https://gronkh.tv/feed',
+        'info_dict': {
+            '_type': 'playlist',
+            'id': 'feed',
+            'title': 'feed',
+            'description': 'feed',
```
[the remark this answers is not in the document]
Comment (hatienl0i261299) [reply yt_dlp/extractor/gronkh.py:61]: Actualy, this IE only support one type for download a list video newest that author of website uploader, so I don't know which title or description to put :D

[the remark this answers is not in the document]
Comment (pukkandan) [reply yt_dlp/extractor/gronkh.py:61]: just id is enough

[the remark this answers is not in the document]
Comment (pukkandan) [reply yt_dlp/extractor/gronkh.py]: this is unreolved

File: yt_dlp/extractor/gronkh.py [hunk 3]
```diff
@@ -41,3 +47,54 @@ def _real_extract(self, url):
...
+        'playlist_mincount': 150,
+    }]
+    _PER_PAGE = 25
+
+    def _fetch_page(self, page):
+        items = traverse_obj(self._download_json(
+            'https://api.gronkh.tv/v1/search', 'vods', query={'offset': self._PER_PAGE * page, 'first': self._PER_PAGE},
+            note=f'Downloading stream video page {page + 1}'), ('results', 'videos', ...))
+        for item in items or []:
+            yield self.url_result(f'https://gronkh.tv/watch/stream/{item.get("episode")}', GronkhIE, item.get("episode"), item.get('title'))
```
Comment (pukkandan) [inline yt_dlp/extractor/gronkh.py:96]:
```suggestion
            yield self.url_result(f'https://gronkh.tv/watch/stream/{item["episode"]}', GronkhIE, item['episode'], item.get('title'))
```
same issue above

**Comment by hatienl0i261299** [comment]
@pukkandan Is it oke for this PR, can we merge?

**Pull request merged**